lib/public_suffix.ex

```diff
@@ -140,9 +204,9 @@ defmodule PublicSuffix do
   defp find_prevailing_normal_rule([], _allowed_rule_types), do: nil
   defp find_prevailing_normal_rule([_ | suffix] = domain_labels, allowed_rule_types) do
     cond do
-      @exact_match_rules[domain_labels] in allowed_rule_types -> domain_labels
+      @exact_match_rules[domain_labels] in allowed_rule_types -> {:normal, domain_labels}
       # TODO: "Wildcards are not restricted to appear only in the leftmost position"
-      @wild_card_rules[["*" | suffix]] in allowed_rule_types -> domain_labels
+      @wild_card_rules[["*" | suffix]] in allowed_rule_types -> {:normal, ["*"] ++ suffix}
```

Comment:
I recommend using `["*" | suffix]` over `["*"] ++ suffix`.  The `++` operator is `O(n)` over the size of the list on the left, whereas `[a | b]` is highly optimized and is always constant time.

In this case, the list of the left has only one element so I doubt there's a perf difference, but as a a matter of habit I think it's best to prefer the `[a | b]` form when it can be used, as it can here.